query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -467,10 +480,39 @@ private void processComputeIfAbsentCommand(final ComputeIfAbsentCommand command,
     */
    private void processComputeIfAbsentCommand(final ComputeIfAbsentCommand command, final InvocationContext ctx, final Object prevValue, final Object computedValue, TransactionContext transactionContext) {
       if (command.isSuccessful()) {
-         processComputes(command, ctx, prevValue, computedValue, transactionContext);
+         processFunctionalCommand(command, ctx, prevValue, computedValue, transactionContext);
       }
    }
-   private void processComputes(AbstractDataWriteCommand command, InvocationContext ctx, Object prevValue, Object computedValue, TransactionContext transactionContext) {
+
+   /**
+    * Indexing management of a ReadWriteKeyCommand
+    *
+    * @param command the ReadWriteKeyCommand
+    * @param ctx the InvocationContext
+    * @param prevValue the previous value on this key
+    * @param transactionContext Optional for lazy initialization, or reuse an existing context.
+    */
+   private void processReadWriteKeyCommand(final ReadWriteKeyCommand command, final InvocationContext ctx, final Object prevValue, TransactionContext transactionContext) {
+      if (command.isSuccessful()) {
+         processFunctionalCommand(command, ctx, prevValue, ctx.lookupEntry(command.getKey()).getValue(), transactionContext);
+      }
+   }
+   /**
+    * Indexing management of a ReadWriteKeyCommand
+    *
+    * @param command the ComputeIfAbsentCommand
+    * @param ctx the InvocationContext
+    * @param prevValue the value before the call
+    * @param resultValue the result of the command call
+    * @param transactionContext Optional for lazy initialization, or reuse an existing context.
+    */
+   private void processReadWriteKeyCommand(final ReadWriteKeyCommand command, final InvocationContext ctx, final Object prevValue, final Object resultValue, TransactionContext transactionContext) {
+      if (command.isSuccessful()) {
+         processFunctionalCommand(command, ctx, prevValue, resultValue, transactionContext);
+      }
+   }
+
+   private void processFunctionalCommand(AbstractDataWriteCommand command, InvocationContext ctx, Object prevValue, Object computedValue, TransactionContext transactionContext) {
```

Comment:
grab value from the context, if `ctx.lookupEntry` returns non-null. If it returns null (that should be rare, but possible), you are doomed right now: you can't get a value from other owner because it has probably already applied the command. You need to invalidate all indexes, IIUC - don't know how to do that, @anistor or @Sanne could give some hints.